### DISCUSSION TITLE
while Passing the initialData after saving it in local storage getting TypeError

### DISCUSSION THREAD

**Discussion by haider000** [excalidraw/excalidraw#4920] [Q&A]
While integrating the Excalidraw in my project I was trying to save the data in local storage but while passing the same saved data of local storage to Excalidraw as initialData I was getting the TypeError: this.state.collaborators.forEach is not a function.

![Screenshot (519)](https://user-images.githubusercontent.com/35297934/158134997-c9525ad0-85ed-4980-870e-074499438dbe.png)

So I found out that while saving the data in local storage the collaborators property is not being saved as a Map while in type definition the appState.collaborators is a map and this was the reason for the above typeError as while passing the initialData Excalidraw expected it to be a Map.

So I removed the property appState.collaborators using filter while saving the data to local storage to overcome this error.

```js
const onChange = (
    elements: readonly ExcalidrawElement[],
    appState: AppState
  ) => {
   debounce(
    (elements: readonly ExcalidrawElement[], state: AppState) => {
      // convert object to array
      const asArray = Object.entries(state)

      // apply filter on array
      const filtered = asArray.filter(([key, _]) => key !== 'collaborators')

      // Convert the key/value array back to an object:
      const appState = Object.fromEntries(filtered) as AppState

      setExcaliState({ elements, appState })
    },
    900
  )
  }

```

**Comment by ad1992** [comment]
Even though we should add a safe check in the package as well. I would suggest to only save the data which is needed and not dump everything into to local storage, this way you will be saving/sending the data which is being used in your app
